it/src/main/java/com/google/cloud/teleport/it/pubsub/DefaultPubsubResourceManager.java

```diff
@@ -0,0 +1,224 @@
+    this.topicAdminClient = topicAdminClient;
+    this.subscriptionAdminClient = subscriptionAdminClient;
+  }
+
+  /** Build a random id to use as resource prefix. */
+  private String buildRandomId() {
+    final byte[] randomBytes = new byte[4];
+    random.nextBytes(randomBytes);
+
+    return BaseEncoding.base32().omitPadding().lowerCase().encode(randomBytes);
+  }
+
+  public static Builder builder(String projectId) {
+    checkArgument(!projectId.isEmpty(), "projectId can not be empty");
+    return new Builder(projectId);
+  }
+
+  @Override
+  public Topic createTopic(String topicName) {
+    LOG.info("Creating topic '{}'", topicName);
+
+    return this.topicAdminClient.createTopic(getTopicName(topicName));
+  }
+
+  @Override
+  public Subscription createSubscription(String topicName, String subscriptionName) {
+    return this.createSubscription(getTopicName(topicName), subscriptionName);
+  }
+
+  @Override
+  public Subscription createSubscription(TopicName topicName, String subscriptionName) {
+    LOG.info("Creating subscription '{}' for topic '{}'", subscriptionName, topicName);
+
+    return this.subscriptionAdminClient.createSubscription(
+        getSubscriptionName(subscriptionName),
+        topicName,
+        PushConfig.getDefaultInstance(),
+        DEFAULT_ACK_DEADLINE_SECONDS);
+  }
+
+  @Override
+  public synchronized void cleanupAll() {
+    LOG.info("Attempting to cleanup manager.");
+
+    // as more resources can exist on the project, we delete only subscriptions for current testId
+    try {
+      ListSubscriptionsPagedResponse subsResponse =
+          subscriptionAdminClient.listSubscriptions(ProjectName.of(this.projectId));
+      if (subsResponse != null) {
+        String deletePrefix = String.format(SUBSCRIPTIONS_PATTERN, projectId, getTestPrefix());
+        for (Subscription subscription : subsResponse.iterateAll()) {
+          if (subscription.getName().startsWith(deletePrefix)) {
+            LOG.info("Deleting subscription '{}'", subscription.getName());
+            subscriptionAdminClient.deleteSubscription(subscription.getName());
+          }
+        }
+      }
+    } finally {
+      this.subscriptionAdminClient.close();
+    }
+
+    // as more resources can exist on the project, we delete only topics for current testId
+    try {
+      ListTopicsPagedResponse topicsResponse =
+          topicAdminClient.listTopics(ProjectName.of(this.projectId));
+      if (topicsResponse != null) {
+        String deletePrefix = String.format(TOPICS_PATTERN, projectId, getTestPrefix());
+        for (Topic topic : topicsResponse.iterateAll()) {
+          if (topic.getName().startsWith(deletePrefix)) {
+            LOG.info("Deleting topic '{}'", topic.getName());
+            topicAdminClient.deleteTopic(topic.getName());
+          }
+        }
+      }
+    } finally {
+      this.topicAdminClient.close();
+    }
+
+    LOG.info("Manager successfully cleaned up.");
+  }
+
+  @VisibleForTesting
```

Comment:
We have tests both validating functionality, as well as using the method for assertions (see testCleanupSubscriptions). Let me know if you still think this is not proper.

Reply:
I think I see that you want to simplify, but my initial idea for passing in a name is to add something meaningful to the test logic, and that may in case ITs need to be debugged.

It may be not too clear on PubsubToTextIT, but for example, there are cases that you could potentially have 3 topics (e.g., "input", "output", "dlq"), and they could be created "test-pubsub-to-pubsub-202208260956-(input|output|dlq)" so that the names are somewhat meaningful rather than just using sequences.

What do you think?

Reply:
Maybe rather than receiving the full topic name, just receiving a suffix if tests need to differentiate their purpose. Just make it clear from the name/documentation how it impacts the actual topic name.

Part of the purpose of this framework is to not think about isolation in the tests themselves, and receiving the full topic name goes against that.

Reply:
`testId` always gets prepended on the topic name for such isolation. I'll make this clearer on the docs/parameter names.

Reply:
It is just the suffix. I've made it clearer on the JavaDoc that the topic/subscription will not be used as-is.

Reply:
Ok, getting back to the original ask: Could you remove these `@VisibleForTesting` annotations? It should be possible to use a captor and assert the captured value is correct rather than testing these methods directly.